### PR TITLE
Af 352 refactor settle liquidation

### PR DESCRIPTION
**Pull request by Thomas-Smets**
First merge https://github.com/arcadia-finance/accounts-v2/pull/59

### CODE REVIEW

**Review by j-vp — APPROVED** [review]
lgtm

File: src/Liquidator.sol [hunk 1]
```diff
@@ -51,7 +51,6 @@ contract Liquidator is Owned, ILiquidator {
 
     // Struct with additional information about the auction of a specific Account.
     struct AuctionInformation {
-        address originalOwner; // The address of the original owner of the Account.
```
Comment (zekiblue) [inline src/Liquidator.sol:54]: I think better to keep this, it cuts one call to account in the happy flow. And it is good data point to query later on

Comment (Thomas-Smets) [reply src/Liquidator.sol:54]: What do you mean with "good data point to query later on"?
Changes of ownership emit events as well.

File: src/LendingPool.sol [hunk 2]
```diff
@@ -905,22 +916,33 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
         processInterests
     {
         (, uint256 auctionTerminationReward, uint256 liquidationFee) = _calculateRewards(startDebt);
 
-        // openDebt equals startDebt + interests + liquidationInitiatorReward + auctionTerminationReward + liquidationFee + interests - bids.
+        // Any remaining debt that was not recovered during the auction must be written off.
+        // Depending on the size of the remaining debt, different stakeholders will be impacted.
         uint256 openDebt = maxWithdraw(account);
         if (openDebt > auctionTerminationReward + liquidationFee) {
+            // "openDebt" is bigger as pending liquidation incentives.
```
Comment (j-vp) [inline src/LendingPool.sol:944]:
```suggestion
            // "openDebt" is bigger than pending liquidation incentives.
```

File: src/LendingPool.sol [hunk 3]
```diff
@@ -905,22 +916,33 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
 
     /**
-     * @notice Settles the liquidation process for a specific Account.
+     * @notice Ends the liquidation process for a specific Account and settles the liquidation incentives/bad debt.
      * @param account The address of the Account undergoing liquidation settlement.
      * @param startDebt The initial debt amount of the liquidated Account.
      * @param terminator The address of the auction terminator.
+     * @dev In the unhappy flow, the auction proceeds are not sufficient to pay out all liquidation incentives
+     *  and maybe not even to pay off all debt. 
+     * @dev The order in which incentives are not payed out/ bad debt is settles is fixed:
```
Comment (j-vp) [inline src/LendingPool.sol:925]:
```suggestion
     * @dev The order in which incentives are not paid out/ bad debt is settles is fixed:
```

File: src/LendingPool.sol [hunk 4]
```diff
@@ -905,22 +916,33 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
+     *   - First, the "liquidationFee", going towards LPs and the Treasury is not payed out.
```
Comment (j-vp) [inline src/LendingPool.sol:926]:
```suggestion
     *   - First, the "liquidationFee", going towards LPs and the Treasury is not paid out.
```

File: src/LendingPool.sol [hunk 5]
```diff
@@ -905,22 +916,33 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
+     *   - Next, the "auctionTerminationReward", going towards the terminator of the auction is not payed out.
```
Comment (j-vp) [inline src/LendingPool.sol:927]:
```suggestion
     *   - Next, the "auctionTerminationReward", going towards the terminator of the auction is not paid out.
```

File: src/LendingPool.sol [hunk 6]
```diff
@@ -905,22 +916,33 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
     {
         (, uint256 auctionTerminationReward, uint256 liquidationFee) = _calculateRewards(startDebt);
 
-        // openDebt equals startDebt + interests + liquidationInitiatorReward + auctionTerminationReward + liquidationFee + interests - bids.
+        // Any remaining debt that was not recovered during the auction must be written off.
+        // Depending on the size of the remaining debt, different stakeholders will be impacted.
         uint256 openDebt = maxWithdraw(account);
         if (openDebt > auctionTerminationReward + liquidationFee) {
+            // "openDebt" is bigger as pending liquidation incentives.
+            // No incentives will be payed out, and a default event is triggered.
```
Comment (j-vp) [inline src/LendingPool.sol:945]:
```suggestion
            // No incentives will be paid out, and a default event is triggered.
```

File: src/LendingPool.sol [hunk 7]
```diff
@@ -931,16 +953,18 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
         } else {
             uint256 remainder;
             if (openDebt >= liquidationFee) {
+                // "openDebt" is bigger as the "liquidationFee" but smaller as the pending liquidation incentives.
```
Comment (j-vp) [inline src/LendingPool.sol:956]:
```suggestion
                // "openDebt" is bigger than the "liquidationFee" but smaller than the pending liquidation incentives.
```

File: src/LendingPool.sol [hunk 8]
```diff
@@ -931,16 +953,18 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
+                // Don't pay out the "liquidationFee" to Lps, partially pay out the "terminator".
                 remainder = (liquidationFee + auctionTerminationReward) - openDebt;
                 realisedLiquidityOf[terminator] += remainder;
             } else {
+                // "openDebt" is smaller as the "liquidationFee" but smaller as the pending liquidation incentives.
```
Comment (j-vp) [inline src/LendingPool.sol:961]:
```suggestion
                // "openDebt" is smaller than the "liquidationFee" but smaller than the pending liquidation incentives.
```

File: src/LendingPool.sol [hunk 9]
```diff
@@ -974,23 +998,23 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
                     realisedLiquidityOf[tranche] -= badDebt;
                 }
                 break;
             } else {
-                //Unhappy flow, should never occur in practice!
-                //badDebt is bigger than balance most junior Tranche -> tranche is completely wiped out
-                //and temporarily locked (no new deposits or withdraws possible).
-                //DAO or insurance might refund (Part of) the losses, and add Tranche back.
+                // Unhappy flow, should never occur in practice!
+                // badDebt is bigger than balance most junior Tranche -> tranche is completely wiped out
```
Comment (j-vp) [inline src/LendingPool.sol:1008]:
```suggestion
                // badDebt is bigger than the balance of most junior Tranche -> tranche is completely wiped out
```

**Review by j-vp — APPROVED** [review]
gtg after comments

**Pull request merged**